From bee7bb3310b356e99e3a0f75f23efbc97f1b0a24 Mon Sep 17 00:00:00 2001
From: Russell Keith-Magee <russell@keith-magee.com>
Date: Mon, 26 Feb 2024 09:21:10 +0800
Subject: [PATCH] config.sub: Include support for arm64_32 and iOS/tvOS/watchOS
 simulators

  * Normalise the watchOS CPU architecture descriptor used by
    Apple (`arm64_32`).
  * Accept the `-simulator` suffix for descriptors for iOS, tvOS and
    watchOS.

The following is the list of compiler triples generated by Apple's
clang toolchains for iOS 12.0, tvOS 9.0, and watchOS 4.0:

  * `arm64-apple-ios12.0`
  * `x86_64-apple-ios12.0-simulator`
  * `arm64-apple-ios12.0-simulator`
  * `arm64-apple-tvos9.0`
  * `x86_64-apple-tvos9.0-simulator`
  * `arm64-apple-tvos9.0-simulator`
  * `arm64_32-apple-watchos4.0`
  * `arm64-apple-watchos4.0-simulator`

The `-simulator` suffix is significant because iOS devices and iOS
simulators have a different ABI.  It is therefore essential that these
can be differentiated in a compiler triple.
---
 config.sub | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index 2c6a07a..1bb6a05 100755
--- a/config.sub
+++ b/config.sub
@@ -4,6 +4,7 @@
 
 # shellcheck disable=SC2006,SC2268 # see below for rationale
 
+# Patched 2024-02-03 to include support for arm64_32 and iOS/tvOS/watchOS simulators
 timestamp='2024-01-01'
 
 # This file is free software; you can redistribute it and/or modify it
@@ -1127,7 +1128,7 @@ case $cpu-$vendor in
 	xscale-* | xscalee[bl]-*)
 		cpu=`echo "$cpu" | sed 's/^xscale/arm/'`
 		;;
-	arm64-* | aarch64le-*)
+	arm64-* | aarch64le-* | arm64_32-*)
 		cpu=aarch64
 		;;
 
@@ -1866,6 +1867,8 @@ case $kernel-$os-$obj in
 		;;
 	*-eabi*- | *-gnueabi*-)
 		;;
+	ios*-simulator- | tvos*-simulator- | watchos*-simulator- )
+		;;
 	none--*)
 		# None (no kernel, i.e. freestanding / bare metal),
 		# can be paired with an machine code file format
-- 
2.47.1

